### PR TITLE
Reformat behave tests

### PR DESCRIPTION
**Pull request by dheyay**
## Why is this needed?

Creates a new tox environment which checks for formatting of gherkin files (behave tests), runs with pre-commit.

Reference for gherkin reformatting library - https://pypi.org/project/reformat-gherkin/

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [ ] No

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,12 +1,16 @@
 repos:
   - repo: https://github.com/ambv/black
-    rev: 22.3.0  # Also stored in dev-requirements.txt; update both together!
+    rev: 24.2.0  # Also stored in dev-requirements.txt; update both together!
```
Comment (renanrodrigo) [inline .pre-commit-config.yaml:3]: The comment says you must update dev-requirements together with this :D

Comment (renanrodrigo) [reply .pre-commit-config.yaml:3]: But besides that, is there a reason to bump black and isort here?

File: dev-requirements.txt [hunk 2]
```diff
@@ -4,3 +4,4 @@ black==22.3.0
 isort==5.12.0
 pre-commit
 shellcheck-py==0.9.0.6
+reformat-gherkin
```
Comment (renanrodrigo) [inline dev-requirements.txt:7]: Should we pin the dependency to a version?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (renanrodrigo) [reply .pre-commit-config.yaml:3]: But the question is : is there a reason to bump those deps?

**Review by renanrodrigo — APPROVED** [review]
Overall LGTM, standardizing this is great, thanks much
I have a couple questions on the dependencies though

File: dev-requirements.txt (hunk 2 above)
Comment (dheyay) [reply dev-requirements.txt:7]: Yes, my bad. I think we should pin it to 3.0.1. Thanks for pointing that out.

**Review by orndorffgrant — APPROVED** [review]
LGTM thank you!

**Pull request merged**